x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/metric/SecurityCacheMetrics.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.metric;
+
+import org.elasticsearch.common.cache.Cache;
+import org.elasticsearch.telemetry.metric.LongWithAttributes;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class SecurityCacheMetrics {
+
+    public static <K, V> List<AutoCloseable> registerAsyncCacheMetrics(MeterRegistry registry, Cache<K, V> cache, CacheType type) {
+        final List<AutoCloseable> metrics = new ArrayList<>();
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                type.metricsPrefix + ".hit.total",
+                "Total number of cache hits.",
+                "count",
+                () -> new LongWithAttributes(cache.stats().getHits())
+            )
+        );
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                type.metricsPrefix + ".miss.total",
+                "Total number of cache misses.",
+                "count",
+                () -> new LongWithAttributes(cache.stats().getMisses())
+            )
+        );
+        metrics.add(
+            registry.registerLongAsyncCounter(
+                type.metricsPrefix + ".eviction.total",
+                "Total number of cache evictions.",
+                "count",
+                () -> new LongWithAttributes(cache.stats().getEvictions())
+            )
+        );
+        metrics.add(
+            registry.registerLongGauge(
+                type.metricsPrefix + ".count.total",
```

Comment:
According to the [naming convention](https://github.com/elastic/elasticsearch/blob/main/modules/apm/NAMING.md) this should not end in `total` and should not be the root of the other metrics. Maybe `.elements.current`?

Reply:
Again, good catch! I think it makes sense to name it as `count.current` since it's closer to the actual cache stats (exposing "current" cache count).